string x 
string y
string rescue_x
string rescue_y
string name